ssl/d1_lib.c

```diff
@@ -373,7 +398,11 @@ int dtls1_handle_timeout(SSL *s)
         return 0;
     }
 
-    dtls1_double_timeout(s);
+    if (s->d1->timer_cb != NULL) {
+        s->d1->timeout_duration_us = s->d1->timer_cb(s, s->d1->timeout_duration_us);
+    } else {
+        dtls1_double_timeout(s);
+    }
```

Comment:
Sorry, didn't spot this on my first read through. OpenSSL style is to not use {} where both the "then" and "else" clauses of an "if" are a single line.